arch/wasm32/atomic_arch.h

```diff
@@ -6,6 +6,13 @@
 
 #define a_barrier() syscall(__NR_membarrier)
 
-#define a_cas(p, t, s) (abort(), s)
+#define a_cas a_cas
```

Comment:
What is the `#define a_cas a_cas` for? AFAIK, having this is identical to not having it.

Reply:
That's needed if you later want to test with `#ifdef`. Just defining a function with that same name will not work - this does not print anything:
```
#include <stdio.h>

int const7()
{
  return 7;
}

int
main()
{
#ifdef const7
  printf("%d\n", const7());
#endif

  return 0;
}
```
But it will work if you add `#define const7 const7`. I don't want to dig into C standards right now but I guess function namespace is separate from preprocessor namespace would be the way to say it more formally?

Reply:
Ah. Alright, looks good to me.

Reply:
While we're at it we may want to add a comment saying it's not really a compare-and-swap - it's not atomic. But then it's ok as long as we're still single threaded. AFAIK WASM MVC does not include atomic operations of this kind so we may want to revisit at a later time. They do [say](https://github.com/WebAssembly/design/blob/master/Portability.md) though that support for CAS is required in the execution environment - they call it compare-and-exchange (apparently it goes by many names).

Reply:
Yea if you don't mind just leaving a comment describing the situation. We should note that there is a threading proposal which will allow us to implement this properly, but without threading it isn't necessary.